### PR TITLE
Druid Resto - fixed numbers for Photo module and overhauled Abundance module

### PR DESCRIPTION
**Pull request by kfinch**
### Description
* Updated Photo numbers to account for recent nerf
* Removed Abundance raw mana numbers calc because it was scuffed.
* Properly separated crit / mana numbers for Abundance because they apply in different cases
* Overhauled the Abundance module with more modern style and with more clear tooltip
* Also added `getOwnBuff` convenience helpers to `Entity` in Core (original versions remain untouched)

### Testing

- Test report URL: `/report/r4bwPWDqV1BYgpad/28-Mythic+Ulgrax+the+Devourer+-+Wipe+4+(4:02)/Youzidurid/standard/statistics`
- Screenshot(s):
image

### CODE REVIEW

File: src/analysis/retail/druid/restoration/modules/spells/Regenesis.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ const TRANQ_MAX_BOOST_PER_RANK = 0.15;
  * **Regenesis**
  * Spec Talent
  *
- * Rejuvenation healing is increased by up to (10 / 20)%, and Tranquility healing is increased by
+ * Rejuvenation healing is increased by up to (15 / 30)%, and Tranquility healing is increased by
```
Comment (kfinch) [inline src/analysis/retail/druid/restoration/modules/spells/Regenesis.tsx:27]: The actual consts here were already correct I'm just updating the tooltip to match

**Comment by kfinch** [comment]
Entity - updated all buff functions to take either an ID or a Spell object

**Comment by kfinch** [comment]
The above change uncovered a bug in DeathRecapTracker, which I have fixed

File: src/analysis/retail/druid/restoration/modules/spells/Abundance.tsx [hunk 2]
```diff
@@ -1,97 +1,157 @@
...
  */
-class Abundance extends Analyzer {
-  manaSavings: number[] = [];
-  critGains: number[] = [];
-  stacks: number[] = [];
-  manaCasts = 0;
+class Abundance extends Analyzer.withDependencies({
+  statTracker: StatTracker,
+  combatants: Combatants,
+}) {
```
Comment (Seriousnes) [inline src/analysis/retail/druid/restoration/modules/spells/Abundance.tsx:30]: Didn't know this existed, very cool

**Review by Seriousnes — APPROVED** [review]

File: src/analysis/retail/druid/restoration/modules/spells/Abundance.tsx (hunk 2 above)
Comment (kfinch) [reply src/analysis/retail/druid/restoration/modules/spells/Abundance.tsx:30]: Yeah it's the new way to do analyzer dependencies that is hopefully cleaner than the old way

**Review by emallson — APPROVED** [review]

**Pull request merged**